content/docs/pulumi-cloud/copilot/api.md

```diff
@@ -0,0 +1,243 @@
+  "messages": [
+    {
+      "role": "user",
+      "kind": "response",
+      "content": "Analyze this update. If it failed, tell me how to resolve the issues."
+    },
+    {
+      "role": "assistant",
+      "kind": "trace",
+      "content": "Conversation with user 'john' in org 'myorg' from console (rest-api-v2)"
+    },
+    {
+      "role": "assistant",
+      "kind": "status",
+      "content": "Executing Pulumi Cloud skill"
+    },
+    {
+      "role": "assistant",
+      "kind": "response",
+      "content": "The update for the stack 'project1/dev' has failed. The failure is due to an error in creating a Virtual Network resource in Azure. The specific error is related to the incorrect handling of an Output<T> type in Pulumi. The error message indicates that calling 'ToString' on an Output<T> is not supported, which leads to a bad request error (HTTP 400) when trying to create the resource.\n\n**How to resolve the issue:**\n- Review the code where the resourceGroupName is being set. Ensure that you are not directly converting an Output<T> to a string using 'ToString'.\n- Use the 'Apply' method or 'Output.Format' to correctly handle the Output<T> type. For example, use `o.Apply(v => $\"prefix{v}suffix\")` or `Output.Format($\"prefix{hostname}suffix\")`.\n- Refer to the Pulumi documentation on [Inputs and Outputs](https://www.pulumi.com/docs/concepts/inputs-outputs) for more guidance on handling Output<T> types properly."
+    }
+  ]
+}
+```
+
+## Post message to conversation
+
+Posts a new message to an existing conversation. The request format is identical to the "Start Conversation", with the addition of the `conversationId` field returned from that call.
+
+```
+POST /api/ai/chat/preview
+```
+
+### Request format
+
+#### Request body schema
+
+```JSON
+{
+    "query": string,
+    "state": {
+        "client": {
+            "cloudContext": {
+                "orgId": string,
+                "url": string
+            }
+        }
+    },
+    "conversationId": string
+}
+```
+
+#### Fields description
+
+- `query`, `orgId`, `url`: Same as in the "Start conversation" request, see above.
+- `conversationId`: conversation to post the message to.
+
+### Response format
+
+Same as in the "Start conversation" request, see above.
+
+### Example request
+
+```bash
+curl -L https://api.pulumi.com/api/ai/chat/preview \
+-H "Authorization: token $PULUMI_ACCESS_TOKEN" \
+-H "Content-Type: application/json" \
+-d '{
+    "query": "Why can I not directly convert an Output<T> to a string?",
+    "state": {
+        "client": {
+            "cloudContext": {
+                "orgId": "myorg",
+                "url": "https://app.pulumi.com/myorg/project1/dev/updates/4"
+            }
+        }
+    },
+    "conversationId":"522e0fdb-e992-4be4-9937-37249ed93289"
+}'
+```
+
+### Example response
+
+````JSON
+{
+  "conversationId": "522e0fdb-e992-4be4-9937-37249ed93289",
+  "messages": [
+    // omitted for brevity
+    // ...
```

Comment:
Ah.. we don't actually return the full conversation here it seems? Just the new messages. Maybe we remove the omitted for brevity bit, I thought that meant it would return all messages.

Reply:
I think it's fine - there are some trace messages that are not important, and we can change them later so I don't want to put them in the docs yet.